CMakeLists.txt

```diff
@@ -1156,6 +1157,29 @@ if(ALLOW_ARM64_CRC32)
     endif()
 endif()
 
+# LoongArch CRC32 Intrinsics are in larchintrin.h.
+# These are supported by at least GCC and Clang.
+option(ALLOW_LOONGARCH_CRC32 "Allow LoongArch CRC32 instruction if \
+supported by the system" ON)
+
+if(ALLOW_LOONGARCH_CRC32)
+    check_c_source_compiles("
+            // As at now only 64-bit LoongArch is supported.
+            #if !(defined(__loongarch__) && __loongarch_grlen >= 64)
+            #   error LoongArch CRC32 only supported on 64-bit LoongArch
+            #endif
+
+            #include <larchintrin.h>
+
+            int my_crc(int word, int crc)
+            {
+                return __crc_w_w_w(word, crc);
+            }
+            int main(void) { return 0; }
+        "
+        HAVE_LOONGARCH_CRC32)
```

Comment:
It looks like the `target_compile_definitions(liblzma PRIVATE HAVE_LOONGARCH_CRC32)` was accidentally omitted during the simplification. Can this be added back?

Reply:
Oops.  Stupid I...

Reply:
Done.